ts/components/screens/BaseScreenComponent/index.tsx

```diff
@@ -0,0 +1,234 @@
+  appLogo,
+  children,
+  contextualHelp,
+  contextualHelpMarkdown,
+  customGoBack,
+  customRightIcon,
+  dark,
+  faqCategories,
+  goBack,
+  headerBackgroundColor,
+  headerBody,
+  headerTitle,
+  isSearchAvailable,
+  onAccessibilityNavigationHeaderFocus,
+  primary,
+  reportAttachmentTypes,
+  shouldAskForScreenshotWithInitialValue,
+  showInstabugChat,
+  titleColor
+}) => {
+  const currentScreenName = useSelector(
+    (store: GlobalState) => getCurrentRouteName(store.nav) || "not-found"
+  );
+
+  const [isHelpVisible, setIsHelpVisible] = useState(false);
+  // if the content is markdown we listen for load end event, otherwise the content is
+  // assumed always loaded
+  const [markdownContentLoaded, setMarkdownContentLoaded] = useState<boolean>(
+    !contextualHelpMarkdown
+  );
+
+  // used to trigger the side-effect base on timeout to take the screenshot
+  const [requestAssistanceData, setRequestAssistanceData] = useState<{
+    payload: RequestAssistancePayload;
+    attachmentConfig: DefaultReportAttachmentTypeConfiguration;
+  } | null>(null);
+
+  useEffect(() => {
+    if (requestAssistanceData) {
+      // since in Android we have no way to handle Modal onDismiss event https://reactnative.dev/docs/modal#ondismiss
+      // we force handling here. The timeout is due to wait until the modal is completely hidden
+      // otherwise in the Instabug screeshot we will see the contextual help content instead the screen below
+      // TODO: To complete the porting to 0.63.x, both iOS and Android will use the timeout. https://www.pivotaltracker.com/story/show/174195300
+      setTimeout(() => {
+        handleOnContextualHelpDismissed(
+          requestAssistanceData.payload,
+          requestAssistanceData.attachmentConfig
+        );
+        setRequestAssistanceData(null);
+      }, ANDROID_OPEN_REPORT_DELAY);
+    }
+  }, [requestAssistanceData]);
+
+  const showHelp = () => {
+    void mixpanelTrack("OPEN_CONTEXTUAL_HELP", {
+      SCREEN_NAME: noAnalyticsRoutes.has(currentScreenName)
+        ? "_"
+        : currentScreenName
+    });
+
+    maybeDark(dark).map(_ =>
+      setStatusBarColorAndBackground("dark-content", customVariables.colorWhite)
+    );
+    setIsHelpVisible(true);
+    setMarkdownContentLoaded(!contextualHelpMarkdown);
+  };
+
+  const hideHelp = () => {
+    maybeDark(dark).map(_ =>
+      setStatusBarColorAndBackground(
+        "light-content",
+        customVariables.brandDarkGray
+      )
+    );
+    setIsHelpVisible(false);
+  };
+
+  const handleOnRequestAssistance = (payload: RequestAssistancePayload) => {
+    // if reportAttachmentTypes is undefined use the default attachment config
+    const attachmentConfig: DefaultReportAttachmentTypeConfiguration = {
+      ...(reportAttachmentTypes ?? defaultAttachmentTypeConfiguration),
+      screenshot: payload.shouldSendScreenshot ?? true
+    };
+    if (payload.supportType === BugReporting.reportType.bug) {
+      setIsHelpVisible(false);
```

Comment:
maybe also this line could be moved in the `useEffect` block?
just before we start the timeout to raise `handleOnContextualHelpDismissed`.

I find that these 2 operation are strictly related: "I want to dismiss the help and I want to raise an event that assistance is requested"
If we change the behavior of the second part ("I want to raise an event that assistance is requested") maybe "I want to dismiss the help" could be not longer valid

thoughts?

Reply:
It makes sense, I've tested locally and it seems to be working well.
The reason why I left it there is that I was mimicking the class component's behaviour as much as possible to avoid breaking the modal's logic. 
Thank you for pointing it out!
https://github.com/pagopa/io-app/pull/3329/commits/5e96383f6a9e3a43917086b3b3f7e9cac26f72ae